server.go

```diff
@@ -608,7 +561,7 @@ func (s *Server) createDefaultClient() {
 	if s.TLS != nil {
 		transport.TLSClientConfig = s.TLS
 	}
-	s.defaultClient = NewInternalHTTPClientFromURI(nil, &ClientOptions{TLS: s.TLS})
+	s.defaultClient = NewInternalHTTPClientFromURI(s.URI, &ClientOptions{TLS: s.TLS})
```

Comment:
Was the `nil` here a bug?

Reply:
yes, the default client could not use nil URI to create a path with.  Although I believe this may be addressed in the resize branch